### PR TITLE
build: extracted UI components to ui-kit in plugin-base-frontend

### LINKED ISSUE #3460: Add processing for styles in plugin-base
According to the [spike](https://github.com/Kotlin/dokka/issues/3423) for decomposing styles.css we need to add some processing of styles from plugin-base.

**Now:**
* plain css styles from `dokka-subprojects/plugin-base/src/main/resources/dokka/styles` are just copied to the resources to html pages

**Should become:**
styles from `dokka-subprojects/plugin-base/src/main/resources/dokka/styles` 
* are written in scss since it provides flexible components organisation and imports
* are concatenated into one output css file

The source `scss` files could be processed via tools like webpack

The source `scss` files should be processed and compiled before/together with `preparedokkaHtmlFrontendFiles` task

### PR DESCRIPTION
**Pull request by berezinant**
Fixes:
#3460
#3461
#3462

The UI component is just a couple of js and css files which are conntected to the markup by bem, if possible
The assets (ts and scss) are compiled by webpack, the results are stored in plugin-base as plain files js and css (both minified and non-minified for the sake of possible customization by users) files and committed to the repo.

### CODE REVIEW

**Review by whyoleg — APPROVED** [review]
LGTM

File: dokka-subprojects/plugin-base-frontend/src/main/components/search/search.tsx [hunk 1]
```diff
@@ -2,73 +2,81 @@
...
+  const [selected, onSelected] = useState<Option>(data[0]);
+  const onChangeSelected = useCallback(
+    (selectItem: SelectItem<unknown> | null) => {
+      if (!selectItem) {
+        return;
+      }
+      const maybeOption: Option = selectItem as Option;
+      window.location.replace(`${(window as IWindow).pathToRoot}${maybeOption.location}?query=${maybeOption.name}`);
+      onSelected(maybeOption);
+    },
```
Comment (berezinant) [inline dokka-subprojects/plugin-base-frontend/src/main/components/search/search.tsx:25]: another hack just to proceed with legacy code

File: dokka-subprojects/plugin-base-frontend/src/main/ui-kit/tabs/index.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    setItem: (key: string, value: string) => {
+      if (!isLocalStorageAvailable) {
+        return;
+      }
+      localStorage.setItem(key, value);
+    },
+  };
+})();
+
+export function initTabs() {
```
Comment (berezinant) [inline dokka-subprojects/plugin-base-frontend/src/main/ui-kit/tabs/index.ts:36]: This is pretty much extracted from dokka-subprojects/plugin-base/src/main/resources/dokka/scripts/platform-content-handler.js

**Review by IgnatBeresnev — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -38,3 +38,6 @@ android.local.properties
 
 core/out/
 runners/**/out/
+
+**/node_modules
+**/build
```
Comment (adam-enko) [inline .gitignore:43]: Belated nitpick: Are these excludes necessary?

`build/` directories are already excluded on line 30, and `node_modules/` is excluded specifically in [dokka-subprojects/plugin-base-frontend/.gitignore](https://github.com/Kotlin/dokka/blob/8e5c63d035ef44a269b8c43430f43f5c8eebfb63/dokka-subprojects/plugin-base-frontend/.gitignore)

Comment (berezinant) [reply .gitignore:43]: thanks, it's really unnecessary, removed it in https://github.com/Kotlin/dokka/pull/3714/files

**Pull request merged**